lib/src/main/java/uk/gov/di/ipv/cri/address/library/service/AddressSessionService.java

```diff
@@ -294,6 +289,67 @@ private boolean validSignature(SignedJWT signedJWT, PublicKey clientPublicKey)
         }
     }
 
+    public List<CanonicalAddressWithResidency> parseAddresses(String addressBody)
+            throws AddressProcessingException {
+        List<CanonicalAddressWithResidency> addresses;
+        try {
+            ObjectMapper mapper = new ObjectMapper();
+            addresses = mapper.readValue(addressBody, List.class);
+        } catch (JsonProcessingException e) {
+            throw new AddressProcessingException(
+                    "could not parse addresses..." + e.getMessage(), e);
+        }
+
+        return addresses;
+    }
+
+    public AddressSessionItem getSession(String sessionId) {
+        return dataStore.getItem(sessionId);
+    }
+
+    public void validateSessionId(String sessionId)
+            throws SessionValidationException, SessionNotFoundException, SessionExpiredException {
+        if (sessionId == null) {
+            throw new SessionValidationException("session id is null");
+        }
+        if (sessionId.isEmpty()) {
+            throw new SessionValidationException("session id is empty");
```

Comment:
I'm conflicted on whether we should even have these guards - as we're meant to be validating for a uuid on the api gateway. @jkunle I've see that you also re-check some stuff that API gateway is checking - we should all have a chat about whether we need to do this more than once.